consensus-types/types/payload.go

```diff
@@ -454,13 +457,15 @@ func (p *ExecutionPayload) UnmarshalJSON(input []byte) error {
 }
 
 // Empty returns an empty ExecutionPayload for the given fork version.
-func (p *ExecutionPayload) Empty(_ uint32) *ExecutionPayload {
-	return &ExecutionPayload{}
+func (p *ExecutionPayload) Empty(forkVersion uint32) *ExecutionPayload {
+	return &ExecutionPayload{
+		version: forkVersion,
+	}
 }
 
 // Version returns the version of the ExecutionPayload.
 func (p *ExecutionPayload) Version() uint32 {
-	return version.Deneb
+	return p.version
```

Comment:
this change requires, similar to others, that:
- we either only use only ctors that duly set version
- we always call SetVersion after we use naked structs